backend/inspirehep/search/api.py

```diff
@@ -185,16 +187,25 @@ def execute(self, *args, **kwargs):
                 LOGGER.exception("Match reference error.", query_string=query_string)
         return results
 
-    def query_for_superuser_or_users(self, query_string):
+    def query_by_user_role(self, query_string):
         if not is_superuser_or_cataloger_logged_in():
             if "_collections" in query_string:
+                collections = current_app.config["NON_PRIVATE_LITERATURE_COLLECTIONS"]
+                if current_user and current_user.is_authenticated:
+                    private_collection_roles = set(current_user.roles) - set(
+                        map(lambda c: c.value, Roles)
```

Comment:
Maybe ` {role.value for role in Roles}`, it would be more pythonic way I believe.